### PR TITLE
Implement flash_area () and screenshot_area ()

### PR DESCRIPTION
**Pull request by donadigo**
Implements `flash_area` and `screenshot_area`. The last method left to implement is `select_area`. Feel free to suggest a different way / keyframes to visualize a flash.

### CODE REVIEW

File: src/ScreenshotManager.vala [hunk 1]
```diff
@@ -40,7 +40,28 @@ namespace Gala
...
+
+			var transition = new Clutter.KeyframeTransition ("opacity");
+			transition.duration = 200;
+			transition.remove_on_complete = true;
+			transition.progress_mode = Clutter.AnimationMode.LINEAR;
+			transition.set_key_frames (keyframes);
+			transition.set_values (values);
+			transition.set_to_value (0.0f);
+			
+			var flash_actor = new Clutter.Actor ();
```
Comment (ricotz) [inline src/ScreenshotManager.vala]: Either reuse this actor next time flash_area is called or make sure it gets removed after the transition has finished.

File: src/ScreenshotManager.vala [hunk 2]
```diff
@@ -56,11 +77,10 @@ namespace Gala
...
 		{
-			warning ("ScreenShotArea not implemented");
-			filename_used = "";
-			success = false;
+			debug ("Taking area screenshot");
 
-			throw new DBusError.FAILED ("ScreenShotArea not implemented");
+			var image = take_screenshot (x, y, width, height);
+			success = save_image (image, filename, out filename_used);
 		}
```
Comment (ricotz) [inline src/ScreenshotManager.vala]: If success is false there should still be an error thrown.

**Comment by donadigo** [comment]
@ricotz changes pushed.

**Comment by Extravert-ir** [comment]
Is there any update on this?

**Pull request closed without merging**